Billing.API/Services/Invoice/InvoiceService.cs

```diff
@@ -22,6 +23,67 @@ public InvoiceService(ILogger<InvoiceService> logger, IOptions<InvoiceProviderOp
         }
 
         public async Task<IEnumerable<InvoiceListItem>> GetInvoices(string clientPrefix, int clientId)
+        {
+            var dt = await GetInvoiceRecords(clientPrefix, clientId);
+
+            return dt.AsEnumerable().Select(dr => new InvoiceListItem(
+                clientPrefix,
+                clientId.ToString(),
+                dr.Field<DateTime>("SendDate").ToDateTimeOffSet(),
+                dr.Field<string>("DocCur"),
+                dr.Field<decimal>("DocTotal").ToDouble(),
+                $"invoice_{dr.Field<DateTime>("SendDate"):yyyy-MM-dd}_{dr.Field<int>("AbsEntry")}.{dr.Field<string>("FileExt")}")).ToList();
+        }
+
+        public async Task<byte[]?> GetInvoiceFile(string clientPrefix, int clientId, int fileId)
+        {
+            var dt = await GetInvoiceRecords(clientPrefix, clientId, fileId);
+
+            var invoice = dt.AsEnumerable().Select(dr => new InvoiceListItem(
+                clientPrefix,
+                clientId.ToString(),
+                dr.Field<DateTime>("SendDate").ToDateTimeOffSet(),
+                dr.Field<string>("DocCur"),
+                dr.Field<decimal>("DocTotal").ToDouble(),
+                $"{dr.Field<string>("trgtPath")}\\{dr.Field<string>("FileName")}.{dr.Field<string>("FileExt")}")).FirstOrDefault();
+
+            if (invoice == null)
+                return null!;
```

Comment:
If I am not wrong, `!` is not necessary anymore.

```suggestion
                return null;
```

Reply:
If I don't use it, I get an error between the datatypes `byte[]` and `byte[]?`

Reply:
I thought that using `byte[]?` was enough to allow `null` 🤷‍♂️

Reply:
apologies, in this line you're right, can be safely removed. (And done)

Reply:
You are right, here can be safely deleted (and done)